### PR TITLE
Dogusata/add progress bar to prompt input

### PR DESCRIPTION
**Pull request by dogusata**
## This PR includes
- Added progress indicator component
- Added progress options to prompt input field
- Fixed right shift bug of send button if the given text is too long without breaks
- Reverting the readable css custom properties

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/components/chat-item/prompt-input/prompt-progress.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        MynahUIGlobalEvents.getInstance().dispatch(MynahEventNames.PROMPT_PROGRESS_ACTION_CLICK, {
+          tabId: this.props.tabId,
+          actionId: action.id,
+          actionText: action.text
+        });
+      }
+    });
+    this.render = this.progressIndicator.render;
+
+    MynahUITabsStore.getInstance().getTabDataStore(this.props.tabId).subscribe('promptInputProgress', (progressData) => {
```
Comment (Jurredr) [inline src/components/chat-item/prompt-input/prompt-progress.ts:37]: Don't we want to extract this 'promptInputProgress' to a static file like we do with MynahEventNames for example?

File: src/main.ts [hunk 2]
```diff
@@ -404,6 +411,19 @@ export class MynahUI {
       }
     });
 
+    MynahUIGlobalEvents.getInstance().addListener(MynahEventNames.PROMPT_PROGRESS_ACTION_CLICK, (data: {
+      tabId: string;
+      actionId: string;
+      actionText?: string;
+    }) => {
+      if (this.props.onChatPromptProgressActionButtonClicked !== undefined) {
```
Comment (Jurredr) [inline src/main.ts:419]: Is it not better to use != instead of !== here?

File: src/styles/components/_progress.scss [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    position: relative;
+    border-radius: inherit;
+    width: 100%;
+    display: block;
+    opacity: 1;
+    transition: var(--mynah-bottom-panel-transition);
+    &.no-content,
+    &.no-content * {
+        opacity: 0;
+        // display: none!important;
```
Comment (Jurredr) [inline src/styles/components/_progress.scss:12]: Probably should get rid of this line just for the sake of cleaning up

**Review by Jurredr — COMMENTED** [review]
Nice work! Left a couple of small comments, nothing major. :)

File: example/src/main.ts [hunk 4]
```diff
@@ -571,6 +589,25 @@ Use \`@\` to mention a file, folder, or method.`
...
+                valueText: '',
+                value: 100,
+                actions: []
+            }
+          });
+
+          mynahUI.updateStore(tabId, {
+            loadingChat: false,
+          });
+          setTimeout(()=>{
```
[the remark this answers is not in the document]
Comment (dogusata) [reply example/src/main.ts:605]: this is the example app, almost everything is to simulate server responses.

File: src/components/chat-item/chat-item-buttons.ts [hunk 5]
```diff
@@ -11,11 +11,11 @@ import { Icon } from '../icon';
 import { ChatItemFormItemsWrapper } from './chat-item-form-items';
 
 export interface ChatItemButtonsWrapperProps {
-  tabId: string;
+  tabId?: string;
```
[the remark this answers is not in the document]
Comment (dogusata) [reply src/components/chat-item/chat-item-buttons.ts:14]: To make them usable also outside the tab-content area. For example feedback form too.

File: src/components/chat-item/prompt-input/prompt-progress.ts (hunk 1 above)
Comment (dogusata) [reply src/components/chat-item/prompt-input/prompt-progress.ts:37]: No, they're already coming automatically from the list of the keys in the tab-store. So they both support IntelliSense and typescript definitions already.

**Review by Jurredr — APPROVED** [review]

**Pull request merged**